### PR TITLE
test(number): refactor number from cypress to playwright

### PR DESCRIPTION
**Pull request by sianford**
### Proposed behaviour

Refactor tests to use Playwright

### Current behaviour

Tests currently use Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run Number tests in Playwright Test Runner to check if the number.pw.tsx file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other *.pw.tsx files have regressed

### CODE REVIEW

File: playwright/components/locators.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export const LABEL = '[data-element="label"]';
 export const COMMON_INPUT = ".common-input__";
 export const BUTTON = 'button[type="button"]';
 export const COMMON_INPUT_PREFIX = '[data-element="textbox-prefix"]';
+export const CHARACTER_COUNT = 'div[data-element="character-count"]';
```
Comment (ZhuoyuJin) [inline playwright/components/locators.ts:20]: For this line and the one below, they are referring to the same thing`'div[data-element="character-count"]',` can we only have one const and delete the other one?

File: src/components/number/components.test-pw.tsx
[the remark this answers is not in the document]
Comment (ZhuoyuJin) [reply src/components/number/components.test-pw.tsx:1]: Added based on Robin's comment, we can also remove those duplicate stories which are added to the `components.test-pw.tsx` from `number-test.stores.tsx`.

File: src/components/number/number.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+
+test.describe("check props", () => {
+  testData.forEach((labelValue) => {
+    test(`should check label renders with ${labelValue} as value`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<NumberInputComponent label={labelValue} />);
+
+      const getDataElementByValueElementLabel = getDataElementByValue(
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/number/number.pw.tsx:62]: Quite right. I changed them to be more meaningful.

File: src/components/number/number.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+  SIZE,
+  VALIDATION,
+} from "../../../playwright/support/constants";
+import { ICON } from "../../../playwright/components/locators";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+
+test.describe("check props", () => {
+  testData.forEach((labelValue) => {
+    test(`should check label renders with ${labelValue} as value`, async ({
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/number/number.pw.tsx:56]: Terminology changed. It was a hang over from the original Cypress test case name.

File: src/components/number/number.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,699 @@
...
+import {
+  CHARACTERS,
+  SIZE,
+  VALIDATION,
+} from "../../../playwright/support/constants";
+import { ICON } from "../../../playwright/components/locators";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+
+test.describe("check props", () => {
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/number/number.pw.tsx:54]: I added that to this "check props"describe.

File: src/components/number/number.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,685 @@
...
+  test("should pass accessibility tests for autofocus example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<AutoFocus />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for with label inline example", async ({
```
Comment (robinzigmond) [inline src/components/number/number.pw.tsx:613]:
```suggestion
  test("should pass accessibility tests for label inline example", async ({
```

**suggestion**. Same in the next 6 tests.

Comment (robinzigmond) [reply src/components/number/number.pw.tsx:613]: turns out this is OK as it's due to the name of the story being "with label inline" etc

**Review by robinzigmond — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**